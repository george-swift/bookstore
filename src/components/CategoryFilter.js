import PropTypes from 'prop-types';
import { CATEGORIES } from '../constants';

export default function CategoryFilter({ filterCategory }) {
  const categories = [...['All'], ...CATEGORIES];

  const handleFilterChange = (e) => {
    const { value } = e.target;
    filterCategory(value);
  };

  return (
    <select
      className="form-select my-3 w-auto"
      onChange={handleFilterChange}
    >
      {
        categories.map((category) => (
          <option key={category} value={category}>
            {category}
          </option>
        ))
      }
    </select>
  );
}

CategoryFilter.propTypes = {
  filterCategory: PropTypes.func.isRequired,
};
