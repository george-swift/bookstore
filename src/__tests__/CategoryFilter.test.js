import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render, within } from '@testing-library/react';
import CategoryFilter from '../components/CategoryFilter';

describe('Testing the category Filter component', () => {
  const changeFilter = jest.fn(() => {});

  it('should render correctly', () => {
    const tree = renderer
      .create(<CategoryFilter filterCategory={changeFilter} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display categories in option elements', () => {
    const { getByText } = render(<CategoryFilter filterCategory={changeFilter} />);
    const select = document.querySelector('.form-select');
    const options = within(select).getAllByRole('option');
    expect(options).toHaveLength(8);
    expect(getByText('All')).toBeInTheDocument();
    expect(getByText('Biography')).toBeInTheDocument();
  });
});
