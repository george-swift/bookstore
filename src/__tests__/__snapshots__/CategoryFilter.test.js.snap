// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the category Filter component should render correctly 1`] = `
<select
  className="form-select my-3 w-auto"
  onChange={[Function]}
>
  <option
    value="All"
  >
    All
  </option>
  <option
    value="Action"
  >
    Action
  </option>
  <option
    value="Biography"
  >
    Biography
  </option>
  <option
    value="History"
  >
    History
  </option>
  <option
    value="Horror"
  >
    Horror
  </option>
  <option
    value="Kids"
  >
    Kids
  </option>
  <option
    value="Learning"
  >
    Learning
  </option>
  <option
    value="Sci-Fi"
  >
    Sci-Fi
  </option>
</select>
`;
